sparrow-py/pysrc/sparrow_py/_timestream.py

```diff
@@ -730,6 +740,53 @@ def lookup(self, key: Union[Timestream, Literal]) -> Timestream:
         """
         return Timestream._call("lookup", key, self)
 
+    def shift_to(self, time: Union[Timestream, datetime]) -> Timestream:
+        """
+        Create a Timestream shifting the time of each point to `time`.
+
+        If multiple values are shifted to the same time, they will be emitted in
+        the order in which they originally occurred.
+
+        Parameters
+        ----------
+        time : Union[Timestream, datetime]
+            The time to shift to.
+            This must be a datetime or a Timestream of timestamp_ns.
+
+        Returns
+        -------
+        Timestream
+            Timestream containing the shifted points.
+        """
+        if isinstance(time, datetime):
+            time_ns = time.timestamp() * 1e9
+            return Timestream._call("shift_to", time_ns, self)
```

Comment:
This doesn't work because it requires a `timestamp_ns` type. Aside from changing the fenl signature, what else could we quickly do here?

Reply:
I think we'll need to implement cast, then we can do:

```
time_ns = time.timestamp() * 1e9
time_ns = Timestream._cast(time_ns, pa.timestamp('ns'))
return Timestream._call("shift_to", time_ns, self)
```